python_modules/dagster/dagster/_core/execution/plan/inputs.py

```diff
@@ -380,43 +357,18 @@ def required_resource_keys(self, job_def: JobDefinition) -> Set[str]:
 
 
 @whitelist_for_serdes(storage_field_names={"node_handle": "solid_handle"})
-class FromStepOutput(
-    NamedTuple(
-        "_FromStepOutput",
-        [
-            ("step_output_handle", StepOutputHandle),
-            ("fan_in", bool),
-            # deprecated, preserved for back-compat
-            ("node_handle", NodeHandle),
-            ("input_name", str),
-        ],
-    ),
-    StepInputSource,
-):
+@record
+class FromStepOutput(StepInputSource):
     """This step input source is the output of a previous step.
     Source handle may refer to graph in case of input mapping.
     """
 
-    def __new__(
-        cls,
-        step_output_handle: StepOutputHandle,
-        fan_in: bool,
-        # deprecated, preserved for back-compat
-        node_handle: Optional[NodeHandle] = None,
-        input_name: Optional[str] = None,
-    ):
-        return super().__new__(
-            cls,
-            step_output_handle=check.inst_param(
-                step_output_handle, "step_output_handle", StepOutputHandle
-            ),
-            fan_in=check.bool_param(fan_in, "fan_in"),
-            # add placeholder values for back-compat
-            node_handle=check.opt_inst_param(
-                node_handle, "node_handle", NodeHandle, default=NodeHandle("", None)
-            ),
-            input_name=check.opt_str_param(input_name, "input_handle", default=""),
-        )
+    step_output_handle: StepOutputHandle
+    fan_in: bool
+
+    # deprecated, preserved for back-compat
+    node_handle: NodeHandle = NodeHandle("", None)
+    input_name: str = ""
```

Comment:
any world in which we have serialized StepInputSources with an input_name of `None` that were previously getting converted by the logic in the __new__ thing but now will not be?

Reply:
good call - worth maintaining behavior ill change these to be like the others where we `input_name or ""`